lighthouse-core/scripts/generate-bundlephobia-database.js

```diff
@@ -0,0 +1,145 @@
+ * @param {string} library
+ * @return {boolean}
+ */
+function hasBeenRecentlyScraped(library) {
+  if (!database[library] || database[library].lastScraped === 'Error') return false;
+  return (Date.now() - database[library].lastScraped) / (1000 * 60 * 60 * 24.0) < 7;
+}
+
+/**
+ * Returns true if the object represents valid BundlePhobia JSON.
+ * The version string must not match this false-positive expression: '{number} packages'.
+ * @param {any} library
+ * @return {boolean}
+ */
+function validateLibraryObject(library) {
+  return library.hasOwnProperty('name') &&
+    library.hasOwnProperty('size') &&
+    library.hasOwnProperty('gzip') &&
+    library.hasOwnProperty('description') &&
+    library.hasOwnProperty('repository') &&
+    library.hasOwnProperty('version') &&
+    !library.version.match(/^([0-9]+) packages$/);
+}
+
+/**
+ * Save BundlePhobia stats for a given npm library to the database.
+ * @param {string} library
+ * @param {number} index
+ */
+async function collectLibraryStats(library, index) {
+  return new Promise(async (resolve, reject) => {
+    console.log(`\n◉ (${index}/${librarySuggestions.length}) ${library} `);
+
+    if (hasBeenRecentlyScraped(library)) {
+      console.log(`   ❕ Skipping`);
+      resolve();
+      return;
+    }
+
+    /** @type {Array<BundlePhobiaLibrary>} */
+    const libraries = [];
+    /** @type {string|number} */
+    let lastScraped = Date.now();
+
+    const versions = await getPackageVersionList(library, 10);
+    for (const version of versions) {
+      try {
+        const libraryJSON = await fetchPackageStats(version);
+        if (validateLibraryObject(libraryJSON)) libraries.push(libraryJSON);
+      } catch (e) {
+        console.log(`   ❌ Failed to fetch stats | ${version}`);
+        lastScraped = 'Error';
+      }
+    }
+
+    for (let index = 0; index < libraries.length; index++) {
+      const library = libraries[index];
+
+      database[library.name] = {
+        ...database[library.name],
+        [library.version]: {
+          gzip: library.gzip,
+        },
+      };
+
+      if (index === 0) {
+        database[library.name] = {
+          repository: library.repository,
+          lastScraped,
+          ...database[library.name],
+        };
+
+        database[library.name]['latest'] = database[library.name][library.version];
+      }
+
+      if (lastScraped === 'Error') {
+        database[library.name] = {
+          ...database[library.name],
+          lastScraped,
+        };
+      }
+
+      console.log(`   ✔ ${library.version}` + (index === 0 ? ' (latest)' : ''));
+    }
+
+    resolve();
+  });
+}
+
+(async () => {
+  const startTime = new Date();
+  console.log(`Collecting ${librarySuggestions.length} libraries...`);
+
+  for (let i = 0; i < librarySuggestions.length; i++) {
+    try {
+      await collectLibraryStats(librarySuggestions[i], i + 1);
+    } catch (e) {
+      console.log('Exiting early...\n');
```

Comment:
the first thing we'll do when this happens is want to know why ;)

```suggestion
      console.log('An error occurred: ${e.stack}\nExiting early...\n');
```

Reply:
bump